server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1583,12 +1583,27 @@ public Volume destroyVolume(long volumeId, Account caller, boolean expunge, bool
                 s_logger.warn("Failed to expunge volume: " + volumeId);
                 return null;
             }
+            markVolumeSnapshotsAsDestroyed(volume);
             removeVolume(volume.getId());
         }
 
         return volume;
     }
 
+    private void markVolumeSnapshotsAsDestroyed(VolumeVO volume) {
+        if (!Storage.ImageFormat.QCOW2.equals(volume.getFormat())) {
```

Comment:
If you have a VM with Ceph's volumes, for example, and `snapshot.backup.to.secondary` is disabled when you expunge the VM, all snapshots remain active in UI, but there are no records in the `snapshot_store_ref` table.

Reply:
@slavkap 
I was not quite sure if the snapshots still work and removing this check will not lead to other unexpected issues.
If we can make sure it is not a problem to mark volume snapshot as destroyed if snapshot does not exist in primary and secondary storage, we can remove this check.

Reply:
Yes, it's better to remove this check because if we don't have snapshots on primary and secondary storage, the snapshot has to be marked as destroyed. I will test it

Reply:
@slavkap 
good.
I have removed this check, and fixed the issue in vm expunge/destroy.